main/ejml-dsparse/test/org/ejml/sparse/csc/mult/TestMatrixMatrixMultWithSemiRing_DSCC.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2009-2020, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.sparse.csc.mult;
+
+import org.ejml.data.DMatrixSparseCSC;
+import org.ejml.ops.DSemiRing;
+import org.ejml.ops.DSemiRings;
+import org.ejml.sparse.csc.CommonOpsWithSemiRing_DSCC;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestMatrixMatrixMultWithSemiRing_DSCC {
```

Comment:
Where is the matrix-matrix multiplication in this test? I would have expected this to test sparse-dense, dense-sparse, and dense-dense matmuls with semi-ring algebras.

Reply:
I created the sparse vector as a matrix, so I somewhat test sparse-sparse matmuls.
Dense-dense is currently not supported, as I only introduced semi-rings for sparse matrices yet.

Sparse-dense and dense-sparse sparse tests could be added of course.
In general it would be nice to have extensive tests for these functions, but I think won't have the time to do so currently.
Would be very glad for any help there!

Reply:
Makes sense. Eventually it would be nice to have dedicated methods dense-dense and dense-sparse/sparse-dense methods for matrix multiplication. But even though DSCC is not optimized for dense matrices, we would still expect a sparse matrix filled with nonzeroes to produce a numerically correct result.

Reply:
@breandan I added some test, also including other operations and dense/sparse version of DSCC.
I came up with an easy way, by just using the PLUS-TIMES semi-ring and asserting against the existing implementation.
I think I can conclude the correctness for other semi-rings this way, without asserting on concrete results.

Reply:
I think the testing can still be more exhaustive, like the `add` Variants of `mult` arent directly tested yet. I think the existing test cover the most vital code paths though.  
I would like to create a ticket for more extensive testing (probably after the PR is merged?).

Reply:
As the semi ring is used on primitve operations I think they can also provide a practical value for dense variations. 
Maybe its enough to wait for someone to open an issue for dense ops or handle it by auto-generating the semi ring version (like you already proposed).